http/parse.go

```diff
@@ -218,7 +218,10 @@ func parseResponse(httpRes *http.Response, req *cmds.Request) (cmds.Response, er
 			return nil, fmt.Errorf("unknown error content type: %s", contentType)
 		default:
 			// handle errors from headers
-			e.Message = httpRes.Header.Get(StreamErrHeader)
```

Comment:
Should we not be *returning* an error in this case?

Reply:
As it stands now the error will be returned on the first call to `Next()`, but I can also make it return immediately. I believe that made the test case more complex, but maybe that's okay.